core/object/object.cpp

```diff
@@ -874,6 +890,25 @@ void Object::set_script_and_instance(const Variant &p_script, ScriptInstance *p_
 	script_instance = p_instance;
 }
 
+void Object::set_custom_type_script(const Variant &p_script) {
+	if (custom_type_script == p_script) {
+		return;
+	}
+
+	Ref<Script> c = p_script;
+	ERR_FAIL_COND_MSG(c.is_null() && !p_script.is_null(), "Parameter should be a reference to a valid script (or null).");
+
+	custom_type_script = p_script;
+
+	// Check if existing script is compatible with underlying custom type script being set here
+	if (!custom_type_script.is_null() && !script.is_null() && script != custom_type_script) {
+		Ref<Script> s = script;
+		if (!s->inherits_script(c)) {
+			ERR_FAIL_MSG(s->get_path() + " is not an extension of " + c->get_global_name());
```

Comment:
It's weird to fail after the object is already in invalid state.

Reply:
Uhm, not sure I understand what you mean. So, this checks if both `custom_type_script` and `script` are present but are different scripts. In this case a second check is performed to see if `script` inherits `custom_type_script`. Only if this fails does it fail.